fms/modules/feedforward.py

```diff
@@ -118,11 +119,27 @@ def __init__(
         )
         self.setup_tp(rank, world_size)
 
-    def colwise_param_names(self) -> List[str]:
-        return ["w1"]
-
-    def rowwise_param_names(self) -> List[str]:
-        return ["w2"]
+    def load_weights(
+        self,
+        tensor_values: Dict[str, torch.Tensor],
+    ):
+        # 1. Grab the weights from tensor_values
+        w1_weight = self._get_sd_weight(tensor_values, ["w1", "weight"])
+        w2_weight = self._get_sd_weight(tensor_values, ["w2", "weight"])
+        if self.use_bias:
+            w1_bias = self._get_sd_weight(tensor_values, ["w1", "bias"])
+            w2_bias = self._get_sd_weight(tensor_values, ["w2", "bias"])
+
+        # 2. Raise exceptions for extra weights in tensor_values
+        if len(tensor_values) > (4 if self.use_bias else 2):
+            raise AttributeError("Unused weight")
```

Comment:
might want to consider adding the actual extra/unused weight names to the error messages